scenario/state.py

```diff
@@ -713,6 +806,23 @@ def pebble_ready_event(self):
             )
         return Event(path=normalize_name(self.name + "-pebble-ready"), container=self)
 
+    def get_notice(
+        self,
+        key: str,
+        notice_type: pebble.NoticeType = pebble.NoticeType.CUSTOM,
+    ) -> BoundNotice:
```

Comment:
Good question. I don't think people should be creating `BoundNotice` objects themselves - you could get something weird like:

```
container = Container("foo")
unbound_notice = Notice(...)
notice = BoundNotice(container=container, notice=notice)
```

Where the notice is bound to the container but not actually _in_ it, and I'm not sure what would happen then.

I had it public because it was a visible return type, but I think it is better private, since people shouldn't be using them or need to do anything like `isinstance(_BoundNotice, n)`.